presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftSingleValue.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.predicate.Marker;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.thrift.interfaces.readers.ColumnReaders;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriter;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriters;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.ThriftRange.Bound.toMarkerBound;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public final class ThriftSingleValue
+{
+    public static final String VALUE_COLUMN_NAME = "value";
+    private final List<ThriftColumnData> columnsData;
```

Comment:
It needs to be a list to support structural types, like arrays, maps, etc. In this case multiple data columns will be returned for a logical column. I briefly mentioned in on the `ColumnWriter` interface.